### PR TITLE
[Improvement-4568][API] When the workflow definition name validation API fails, the system prompts that the workflow definition name already exists

### LINKED ISSUE #4568: [Improvement][API] When the workflow definition name validation API fails, the system prompts that the workflow definition name already exists
**Describe the question**
When the workflow definition name validation API fails, the system prompts that the workflow definition name already exists,
the system normally prompts workflow definition name validation error

**Which version of DolphinScheduler:**
 -[1.3.4-release,dev]

### PR DESCRIPTION
**Pull request by zhuangchong**
## What is the purpose of the pull request

this PR closes #4568 

When the workflow definition name validation API fails, the system prompts that the workflow definition name already exists,
the system normally prompts workflow definition name validation error

## Verify this pull request

This change added tests and can be verified as follows:
  - *Manually verified the change by testing locally.*

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/enums/Status.java [hunk 1]
```diff
@@ -133,7 +133,7 @@
     QUERY_TASK_INSTANCE_LOG_ERROR(10103, "view task instance log error", "查询任务实例日志错误"),
     DOWNLOAD_TASK_INSTANCE_LOG_FILE_ERROR(10104, "download task instance log file error", "下载任务日志文件错误"),
     CREATE_PROCESS_DEFINITION(10105, "create process definition", "创建工作流错误"),
-    VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR(10106, "verify process definition name unique error", "工作流定义名称已存在"),
```
Comment (chengshiwen) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/enums/Status.java:136]: There is essentially no difference between the `VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR` and `PROCESS_DEFINITION_NAME_EXIST`, and it's not necessary to add an additional error code.
The easiest way is just renaming `VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR ` to another

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 2]
```diff
@@ -495,7 +495,7 @@ private String getResourceIds(ProcessData processData) {
         if (processDefinition == null) {
             putMsg(result, Status.SUCCESS);
         } else {
-            putMsg(result, Status.VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR, name);
+            putMsg(result, Status.PROCESS_DEFINITION_NAME_EXIST, name);
```
Comment (chengshiwen) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:498]: Replace `VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR` with `PROCESS_DEFINITION_NAME_EXIST ` and ignore `VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR` in other references, so it leads to failing the unit test.
Maybe you should consider all references of the `VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR` globally

Comment (zhuangchong) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:498]: the ut error has been fixed.

Comment (chengshiwen) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:498]: Now `VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR` is use less, so it's better to remove it

Comment (zhuangchong) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:498]: VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR state is used for prompt verifyProcessDefinitionName method throws an exception

```
@ApiException(VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR)
    public Result verifyProcessDefinitionName(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
                                              @ApiParam(name = "projectName", value = "PROJECT_NAME", required = true) @PathVariable String projectName,
                                              @RequestParam(value = "name", required = true) String name) {
        logger.info("verify process definition name unique, user:{}, project name:{}, process definition name:{}",
            loginUser.getUserName(), projectName, name);
        Map<String, Object> result = processDefinitionService.verifyProcessDefinitionName(loginUser, projectName, name);
        return returnDataList(result);
    }

```

PROCESS_DEFINITION_NAME_EXIST state is the prompt used to verify the existence of the workflow definition name

```
ProcessDefinition processDefinition = processDefineMapper.verifyByDefineName(project.getId(), name);
        if (processDefinition == null) {
            putMsg(result, Status.SUCCESS);
        } else {
            putMsg(result, Status.PROCESS_DEFINITION_NAME_EXIST, name);
        }

 ```

I think the VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR and PROCESS_DEFINITION_NAME_EXIST states are both required.

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**